src/MonodomainSolver/FESolver.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef FE_SOLVER_H
+#define FE_SOLVER_H
+
+#include "../utils.hpp"
+
+using namespace dealii;
+
+class FESolver {
+    // Physical dimension (1D, 2D, 3D)
+    static constexpr int dim = 3;
```

Comment:
I notice you define this `dim` variable in all classes. It would seem natural that there is an underlying assumption that all interacting classes share the same value of `dim`, although this assumption is never verified in practice.

I think this is one of the rare situations where a global variable would work best (wrapped in a dedicated namespace just to be sure).